File: program/check-valid-date/CheckValidDate.java
import java.time.LocalDate;
import java.time.format.DateTimeParseException;
import java.util.Scanner;

public class CheckValidDate {

    public static boolean isDateValid(String input) {
        try {
            LocalDate.parse(input);
            return true;
        } catch (DateTimeParseException e) {
            return false;
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.println("Enter a date in the format 'yyyy-mm-dd':");
        String input = scanner.nextLine();

        if (isDateValid(input)) {
            System.out.println("True");
        } else {
            System.out.println("False");
        }
    }
}

File: program/check-valid-date/CheckValidDate.kt
import java.util.Scanner
import java.time.format.DateTimeFormatter

fun main() {
    print("Enter valid date format in yyyy-mm-dd : ")
    println()
    var input = readLine()

    try {
        println("Is $input valid date?: " + CheckValidDate(input!!))
    } catch (exception: Exception) {
        println("Is $input valid date?: " + false)
    }

}

private fun CheckValidDate(date: String): Boolean {
    val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd")
    val formatted = formatter.parse(date)

    return true
}
